data-extraction/pyDataExtraction/commonTypes/Graph.py

```diff
@@ -0,0 +1,76 @@
+from typing import Union, Dict, Optional
+from pyDataExtraction.commonTypes.base import DataType
+
+# from docs
+"""
+interface NodeGraphData {
+	id: string;
+	label?: string;
+	color?: string;
+	shape?: "ellipse" | "box";
+}
+"""
+"""
+interface EdgeGraphData {
+	from: string;
+	to: string;
+	label?: string;
+	id?: string;
+	color?: string;
+	dashes?: boolean;
+}
+"""
+# NOTE: may not be able to encapsulate edge in separate class due to from being a syntax token
+"""
+class Edge:
+    
+    def __init__(self,from: str, to: str,):
+        self.fromnode
+        
+    def __repr__(self):
+        pass
+"""
+# NOTE: ran into issue Node object is not json serializable when ecapsulating in own class
+"""
+class Node(DataType):
+    def __init__(self, id: Union[int, str], label: Optional[str] = None):
+        super().__init__()
+        self.id = id
+        if label is None:
+            self.label = id
+        else:
+            self.label = label
+"""
+
+
+class Graph(DataType):
+    """
+
+
+    Args:
+        DataType (Union[Dict[str,list],Dict[int,list]]): 
+        either expects a dictionary with a list as values or a 2d array
+        some representation of a basic graph 
+    """
+
+    def __init__(self, graph_data: Union[Dict[str, list], Dict[int, list]]):
+        super().__init__()
+        self.kind["graph"] = True
+        # TODO get working for both a dictionary and an nxn array
+        # self.nodes = [Node(node) for node in graph data]
+        self.nodes = []
+        self.edges = []
+        if isinstance(graph_data, dict):
+            # self.nodes = [Node(node) for node in graph data]
+            for node in graph_data:
+                self.nodes.append({"id": str(node)})
+                # TODO change prints to log statements
```

Comment:
It seems like a too minor change to left a TODO for it and not implement it right away

Reply:
that's partially because my experience implementing logs across a library is none, I've only implemented them in a single file before, and didn't try to make it where the log statement was different across classes.